.docker.test.py

```diff
@@ -1,10 +1,21 @@
 import logging
+import os
 
 from dmoj import judgeenv
 from dmoj.citest import ci_test
 from dmoj.executors import get_available
 
-ALLOW_FAIL = {'GASARM', 'JAVA9', 'JAVA10', 'OBJC'}
+arch = os.getenv('ARCH', 'amd64')
```

Comment:
Why not use `platform.machine()`? You can't screw that up.

Reply:
I feel this is less easy to screw up, since the calling script already defines this environment variable for determining which image to use, so there's a clear 1:1 mapping here.

Reply:
I disagree with the assertion that the calling script is reliable. This script is supposed to be generic "test in the docker image", not linked to Travis as a platform. It is worth noting that you have to explicitly pass `-e ARCH="$ARCH"` to `docker run` for this to work.